scripts/fix_headline_data_model.py

```diff
@@ -0,0 +1,358 @@
+
+    if dry_run:
+        logger.info("Skipping re-index in dry run mode")
+        return {"total": total, "success": 0, "errors": 0}
+
+    success_count = 0
+    error_count = 0
+
+    # Process in batches
+    for i in range(0, total, batch_size):
+        batch = users[i : i + batch_size]
+        batch_num = i // batch_size + 1
+        total_batches = (total + batch_size - 1) // batch_size
+        logger.info(f"Re-indexing batch {batch_num}/{total_batches}")
+
+        for user in batch:
+            try:
+                # Update UserDocument
+                user_doc = UserDocument()
+                user_doc.update(user, action="index")
+
+                # Update PersonDocument if user has author profile
+                if hasattr(user, "author_profile") and user.author_profile:
+                    person_doc = PersonDocument()
+                    person_doc.update(user.author_profile, action="index")
+
+                success_count += 1
+                logger.debug(f"Successfully re-indexed user {user.id}")
+
+            except Exception as e:
+                logger.error(f"Error re-indexing user {user.id}: {e}")
+                error_count += 1
+
+    summary = {"total": total, "success": success_count, "errors": error_count}
+
+    logger.info("Re-indexing Summary:")
+    logger.info(f"  Total users: {total}")
+    logger.info(f"  Success: {success_count}")
+    logger.info(f"  Errors: {error_count}")
+
+    return summary
+
+
+def verify_fix():
+    """Verify that the fix was successful by checking for any remaining object headlines."""  # noqa: E501
+    logger.info("=== Verifying Fix ===")
+
+    # Check for any remaining object headlines
+    remaining_objects = 0
+    for author in Author.objects.exclude(headline__isnull=True).iterator():
+        if isinstance(author.headline, dict):
+            remaining_objects += 1
+            logger.warning(f"Author {author.id} still has object headline")
+
+    if remaining_objects == 0:
+        logger.info("✅ All headlines are now in string format")
+        return True
+    else:
+        logger.warning(f"⚠️  {remaining_objects} authors still have object headlines")
+        return False
+
+
+def main():
+    """Main function to run the headline data model fix."""
+    parser = argparse.ArgumentParser(
+        description="Fix headline data model issue",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog="""
+Examples:
+  python fix_headline_data_model.py --dry-run
+  python fix_headline_data_model.py --batch-size 50
+  python fix_headline_data_model.py --skip-reindex
+  python fix_headline_data_model.py --analyze-only
+        """,
+    )
+
+    parser.add_argument(
+        "--dry-run", action="store_true", help="Run in dry-run mode (no changes made)"
+    )
+    parser.add_argument(
+        "--skip-reindex", action="store_true", help="Skip re-indexing step"
+    )
+    parser.add_argument(
```

Comment:
Can we add a param to start at specific ID? Many times these scripts stop in the middle